core/pkg/hostsensorutils/log_coupling.go

```diff
@@ -0,0 +1,45 @@
+package hostsensorutils
+
+import (
+	"fmt"
+)
+
+type LogsMap struct {
+	usedLogs map[string]int
+}
+
+// NewLogCoupling return an empty LogsMap struct ready to be used.
+func NewLogCoupling() *LogsMap {
+	return &LogsMap{
+		usedLogs: make(map[string]int),
+	}
+}
+
+// Update add the logContent to the internal map
+// and set the occurrencty to 1 (if it has never been used before),
+// increment its values otherwise.
+func (lm *LogsMap) Update(logContent string) {
+	_, ok := lm.usedLogs[logContent]
+	if !ok {
+		lm.usedLogs[logContent] = 1
+		fmt.Println(lm)
```

Comment:
Please call `logger.L()` and not `fmt.Println`

Reply:
`fmt.Println` was a typo. I'm going to remove it.